File: ApiRest/ApiRest-COCO-TRIP/ApiRest-COCO-TRIP/Datos/Entity/Grupo.cs
namespace ApiRest_COCO_TRIP.Datos.Entity
{
  /// <summary>
  /// Entidad que almacena los datos asociados a los Grupos de Amigos
  /// </summary>
  public class Grupo: Entidad
  {
    private string nombre; //Nombre del grupo
    private string rutaFoto; //Ruta de la foto
    private byte[] contenidoFoto; //Bytes de la foto
    private int lider; //ID del creador del grupo
    private int cantidadIntegrantes; //Cantidad de miembros del grupo

    /// <summary>
    /// Getters y Setters del atributo Nombre
    /// </summary>
    public string Nombre
    {
      get { return nombre;  }
      set { nombre = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo RutaFoto
    /// </summary>
    public string RutaFoto
    {
      get { return rutaFoto; }
      set { rutaFoto = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo ContenidoFoto
    /// </summary>
    public byte[] ContenidoFoto
    {
      get { return contenidoFoto;  }
      set { contenidoFoto = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Lider
    /// </summary>
    public int Lider
    {
      get { return lider; }
      set { lider = value; }
    }

    /// <summary>
    /// Getters y Setters del atributo Cantidad Integrantes
    /// </summary>
    public int CantidadIntegrantes
    {
      get { return cantidadIntegrantes; }
      set { cantidadIntegrantes = value; }
    }
  }
}

File: FrontOffice/src/dataAccessLayer/domain/grupo.ts
import { Entidad } from './entidad';
//****************************************************************************************************//
//*****************************************CLASE MENSAJE MODULO 3*************************************//
//****************************************************************************************************//

/**
 * Autores:
 * Joaquin Camacho
 * Jose Herrera
 * Sabina Quiroga
 */

/**
 * Descripcion de la clase:
 * Entidad que contiene los datos de los grupos de amigos
 */
export class Grupo extends Entidad
{
    private Nombre: string; //Nombre del grupo
    private RutaFoto: string; //Ruta de la foto
    private ContenidoFoto: any[]; //Bytes de la foto (para cargar)
    private Lider: number; //ID del creador del grupo
    private CantidadIntegrantes: number; //Cantidad de miembros del grupo

    constructor () //Por ahora vacio
    {
        super();
    } 

    /**
     * Retorna el nombre del grupo
     */
    get getNombre() : string 
    {
        return this.Nombre;
    }

    /**
     * Establece el nombre del grupo
     */
    set setNombre(nombre : string) 
    {
        this.Nombre = nombre;
    }

    /**
     * Retorna la ruta de la foto (del servicio web)
     */
    get getRutaFoto() : string
    {
        return this.RutaFoto;
    }

    /**
     * Establece la ruta de la foto (del servicio web)
     */
    set setRutaFoto(rutaFoto : string)
    {
        this.RutaFoto = rutaFoto;
    }

    /**
     * Retorna los bytes de la foto
     */
    get getContenidoFoto() : any[]
    {
        return this.ContenidoFoto;
    }

    /**
     * Establece los bytes de la foto (para guardar en el servicio web)
     */
    set setContenidoFoto(contenidoFoto : any[])
    {
        this.ContenidoFoto = contenidoFoto;
    }

    /**
     * Retorna el identificador unico del lider del grupo
     */
    get getLider() : number
    {
        return this.Lider;
    }

    /**
     * Establece el identificador unico del lider del grupo
     */
    set setLider(idLider : number)
    {
        this.Lider = idLider;
    }

    /**
     * Retorna la cantidad de miembros del grupo de amigos
     */
    get getCantidadIntegrantes() : number
    {
        return this.CantidadIntegrantes;
    }

    /**
     * Establece la cantidad de miembros del grupo de amigos
     */
    set setCantidadIntegrantes(cantidadIntegrantes : number)
    {
        this.CantidadIntegrantes = cantidadIntegrantes;
    }
    
}